docs/languages/python.md

```diff
@@ -385,3 +385,70 @@ faas-cli publish --filter fn1\
     --build-arg UPGRADE_PACKAGES=true
 ```
 
+## OpenTelemetry zero-code instrumentation
+
+Using [OpenTelemetry zero-code instrumentation](https://opentelemetry.io/docs/zero-code/python/) for python functions requires some minor modifications to the existing Python templates.
+
+There are two ways to [customise a template](/cli/templates/#how-to-customise-a-template):
+
+- Fork the template repository and modify the template. Recommended method that allows for distribution and reuse of the template.
+- Pull the template and apply patches directly in the `./template/<language_name>` directory. Good for quick iteration and experimentation with template modifications. The modified template can not be shared and reused. Changes may get overwritten when pulling templates again.
+
+Add the required packages for auto instrumentation to the `requirements.txt` file of the template:
+
+```
+opentelemetry-distro
+opentelemetry-exporter-otlp
+```
+
+Update the Dockerfile to run the bootstrap command after the the template and function packages have been installed:
+
+```diff
+# Build the function directory and install any user-specified components
+USER app
+
+RUN mkdir -p function
+RUN touch ./function/__init__.py
+WORKDIR /home/app/function/
+COPY --chown=app:app function/requirements.txt	.
+RUN pip install --no-cache-dir --user -r requirements.txt
++ RUN opentelemetry-bootstrap -a install
+```
+
+The `opentelemetry-bootstrap -a install` command reads through the list of packages installed in your active site-packages folder, and installs the corresponding instrumentation libraries for these packages, if applicable. The OpenTelemetry Python agent uses [monkey patching](https://stackoverflow.com/questions/5626193/what-is-monkey-patching) to modify functions in these libraries at runtime.
+
+
+Update the fprocess ENV in the Dockerfile to start the OpenTelemetry agent:
+
+```diff
+# configure WSGI server and healthcheck
+USER app
+
+- ENV fprocess="python index.py"
++ ENV fprocess="opentelemetry-instrument python index.py"
+```
+
+Use your modified template to create a new function.
+
+The OpenTelemetry agent can be configured using environment variables on the function:
+
+```yaml
+functions:
+  greet:
+    lang: python3-http-otel
+    handler: ./greet
+    image: greet:latest
```

Comment:
`greet:latest` is probably going to be copy/pasted and someone will find it doesn't work

Reply:
What would you suggest doing instead. There are several other places in the language docs where we use a `stack.yaml` example like this?

- Use a `diff` instead to only highlight the relevant/changed fields?
- Omit the image?

Reply:
Yes use the diff syntax to show what you added..